### ISSUE TITLE
Set '--no-ssh-key' as default bahavior for 'az aks create' in CLI 

### ISSUE THREAD

**Issue by CocoWang-wql** [Azure/azure-cli#27181]
### Preconditions

- [X] No need to upgrade Python SDK or the Python SDK is ready.

### Related command

`az aks create -g <aks resource group> -n <aks name> --generate-ssh-keys`
`az aks create -g <aks resource group> -n <aks name> --no-ssh-key`
`az aks update -g <aks resource group> -n <aks name> --ssh-key-value`

### Resource Provider

Microsoft.ContainerService/Azure Kubernetes Service

### Description of Feature or Work Requested

VMSS requires ssh key for creation. 
During az aks creation, there are several options about ssh key:

1. `--generate-ssh-keys-keys`: az cli will automatically generate a set of ssh keys for the user and put them in the specified directory.
1. No ssh-related parameters:   az cli will use ssh keys in the ~/.ssh/ directory without any SSH-related parameters assigned.
1. `--ssh-key-value` : The user can use `--ssh-key-value` for assigning a dedicated key 
1.  `--no-ssh-key`:  AKS RP will automatically generate a set of ssh key, but it will not be returned to the user. So users can't use this key to access the nodes. As a workaround, they can run `az aks update -g <aks resource group> -n <aks name> --ssh-key-value` to assign new key then ssh.

**Behavior change:**
For the following two reasons:
1) Above 50% clusters are running without ssh key, which means most users prefer not assigning keys.
2)'no ssh key' is safer as there is no risk of revealing private key.

We will set `--no-ssh-key` as the default behavior in CLI. 
Whenever users want to access the nodes, they can run  `az aks update -g <aks resource group> -n <aks name> --ssh-key-value` to assign new keys

### Minimum API Version Required

2023-01-01

### Swagger PR link / SDK link

N/A
We will make this change in GA and no PR link currently

### Target Date

2023-09-01

### PM Contact

qiaolwan

### Engineer Contact

binxi

**Comment by yonzhan** [comment]
Thank you for opening this issue, we will look into it.

**Comment by navba-MSFT** [comment]
Non-customer reported issue. Adding Service team to look into this.

**Comment by FumingZhang** [comment]
Sounds like a breaking change, should keep the change until the next breaking change window (MS Ignite 23).

**Comment by FumingZhang** [comment]
Oops, forgot about this, setting the target date to the next breaking change window (MS Build 24).
